tests/unittest/test_breakpad_resource.py

```diff
@@ -367,13 +360,93 @@ def text_extract_payload_kvpairs_and_json(self, request_generator, metricsmock):
             method="POST", path="/submit", headers=headers, body=data
         )
 
-        bsp = BreakpadSubmitterResource(config=self.empty_config, crashmover=crashmover)
+        bsp = BreakpadSubmitterResource(
+            config=EMPTY_CONFIG, crashmover=FakeCrashMover()
+        )
         with metricsmock as metrics:
             result = bsp.extract_payload(req)
             assert result == ({}, {})
             assert metrics.has_record(stat="malformed", tags=["reason:has_json_and_kv"])
 
+
+@pytest.mark.parametrize(
+    "raw_crash, expected",
+    [
+        ({}, {"metadata": {"collector_notes": []}}),
+        (
+            {"TelemetryClientId": "ou812"},
+            {
+                "metadata": {
+                    "collector_notes": ["Removed TelemetryClientId from raw crash."]
+                }
+            },
+        ),
+        (
+            {"TelemetryServerURL": "ou812"},
+            {
+                "metadata": {
+                    "collector_notes": ["Removed TelemetryServerURL from raw crash."]
+                }
+            },
+        ),
+    ],
+)
+def test_cleanup_crash_report(raw_crash, expected):
+    bsp = BreakpadSubmitterResource(config=EMPTY_CONFIG, crashmover=FakeCrashMover())
+    bsp.cleanup_crash_report(raw_crash)
+    assert raw_crash == expected
+
+
+def test_get_throttle_result(client):
+    raw_crash = {"ProductName": "Firefox", "ReleaseChannel": "nightly"}
+
+    bsp = BreakpadSubmitterResource(config=EMPTY_CONFIG, crashmover=FakeCrashMover())
+    assert bsp.get_throttle_result(raw_crash) == (ACCEPT, "is_nightly", 100)
+
+
+class TestBreakpadSubmitterResourceIntegration:
+    def test_submit_crash_report(self, client):
```

Comment:
This test shows submitting something, then loading the crash data from crash storage and asserting things against it.